truffleruby/src/main/ruby/core/truffle/ffi/pointer.rb

```diff
@@ -183,14 +294,605 @@ def autorelease=(val)
       raise PrimitiveFailure, "FFI::Pointer#autorelease= primitive failed"
     end
 
+    ##
+    # Returns true if autorelease is enabled, otherwise false.
+    def autorelease?
+      Truffle.primitive :pointer_autorelease_p
+      raise PrimitiveFailure, "FFI::Pointer#pointer_autorelease_p primitive failed"
+    end
```

Comment:
We could save around 100 lines of Ruby code by making all of these Java `@CoreMethod` since they have no fallback code.